vault/src/execution.rs

```diff
@@ -149,18 +151,31 @@ impl Request {
             [] => {} // no return-to-self
             [address] => {
                 // one return-to-self address, make sure it is registered
-                let vault_id = provider.get_account_id().clone();
-                let wallet = provider.get_vault(vault_id).await?.wallet;
+                let vault_id = btc_parachain.get_account_id().clone();
+                let wallet = btc_parachain.get_vault(vault_id).await?.wallet;
                 if !wallet.has_btc_address(&address) {
                     info!("Registering address {}", address);
-                    provider.register_address(address.clone()).await?;
+                    (|| async {
```

Comment:
Can you add a comment explaining why we retry this?

Reply:
If the nonce is outdated, it will fail to process the redeem. I added a retry wrapper which we can also use in other places.

Reply:
Doesn't it make sense to hide the retry on outdated nonces inside the `runtime/rpc.rs` code?

Reply:
Possibly, but would be part of a larger refactoring. We can do that later